app/controllers/vendors_controller.rb

```diff
@@ -23,9 +26,22 @@ def create
     end
   end
 
+  def update
+    @vendor = Vendor.find(params[:id])
+    if @vendor.approve
```

Comment:
`#approve` will always return true, right? If so, you shouldn't have this `if` statement here - that implies that sometimes `approve` will return true and sometimes return false.